canonical-web-and-design/maas-ui
# add hardware sync docs links

## Done

- add hardware sync docs links
- move cloud-init content under its checkbox
- update react-components to v0.35.0
  - update snapshots
  - migrate `FilesystemsTable` test to testing-library (enzyme was throwing a Portal error incorrectly after upgrade to latest `react-components`, it made more sense to migrate than to try and debug this)  
    - add `aria-label` to TableActions button (prior to this table actions buttons had no label)
  - migrate `TableMenu` to testing-library

if you prefer this to be broken down, please let me know - but I did think it makes sense to keep it all in a single PR as it's quite small and inter-connected.

## Screenshots
![image](https://user-images.githubusercontent.com/7452681/164234547-27be712c-0382-41a5-aa89-f214c645cc17.png)
![image](https://user-images.githubusercontent.com/7452681/164234568-fc3b8014-52cc-4ae0-b16e-28f94f6f921e.png)
![image](https://user-images.githubusercontent.com/7452681/164274356-a59a3ee7-de0d-4a62-afd4-0ebd43cb85dc.png)


## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

Verify the correct link is displayed on pages below (bolla URLs):
- `/MAAS/r/machine/xxn6s7/summary`
  - The link is displayed in the tooltip under "VIRTUAL MACHINE STATUS"
  - Verify the tooltip remains open on click until dismissed (by clicking outside or pressing "Esc" key")
- `/MAAS/r/machine/sdqtg4/summary`
  - Select Take action -> Deploy... 
  - The link is displayed under "Customise options"

## Fixes

Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/787
Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/827
Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/828
Fixes: https://github.com/canonical-web-and-design/maas-ui/issues/3826
Fixes: https://github.com/canonical-web-and-design/maas-ui/issues/3827

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
